Signal/src/ViewControllers/MessageComposeTableViewController.m

```diff
@@ -3,435 +3,408 @@
-    MessageComposeTableViewControllerSectionContacts,
-    MessageComposeTableViewControllerSection_Count // meta section
-};
-
-NSString *const MessageComposeTableViewControllerCellContact = @"ContactTableViewCell";
+#pragma mark -
 
 @implementation MessageComposeTableViewController
 
-- (nullable instancetype)initWithCoder:(NSCoder *)aDecoder
+- (void)loadView
 {
-    self = [super initWithCoder:aDecoder];
-    if (!self) {
-        return self;
-    }
-
-    _contactsManager = [Environment getCurrent].contactsManager;
-    _phoneNumberAccountSet = [NSMutableSet set];
-
-    _blockingManager = [OWSBlockingManager sharedManager];
-    _blockedPhoneNumbers = [_blockingManager blockedPhoneNumbers];
-
-    [self observeNotifications];
-
-    return self;
-}
-
-- (instancetype)init
+    [super loadView];
+
+    _contactsViewHelper = [ContactsViewHelper new];
+    _contactsViewHelper.delegate = self;
+    _nonContactAccountSet = [NSMutableSet set];
+
+    UIImage *newGroupImage = [UIImage imageNamed:@"btnGroup--white"];
+    OWSAssert(newGroupImage);
+    UIBarButtonItem *newGroupButton = [[UIBarButtonItem alloc] initWithImage:newGroupImage
+                                                                       style:UIBarButtonItemStylePlain
+                                                                      target:self
+                                                                      action:@selector(showNewGroupView:)];
+    newGroupButton.accessibilityLabel
+        = NSLocalizedString(@"CREATE_NEW_GROUP", @"Accessibility label for the create group new group button");
+    self.navigationItem.rightBarButtonItem = newGroupButton;
+
+    // Search
+    UISearchBar *searchBar = [UISearchBar new];
+    _searchBar = searchBar;
+    searchBar.searchBarStyle = UISearchBarStyleMinimal;
+    searchBar.delegate = self;
+    searchBar.placeholder = NSLocalizedString(@"SEARCH_BYNAMEORNUMBER_PLACEHOLDER_TEXT", @"");
+    searchBar.backgroundColor = [UIColor whiteColor];
+    [searchBar sizeToFit];
+
+    _tableViewController = [OWSTableViewController new];
+    _tableViewController.delegate = self;
+    [self.view addSubview:self.tableViewController.view];
+    [_tableViewController.view autoPinWidthToSuperview];
+    [_tableViewController.view autoPinEdgeToSuperviewEdge:ALEdgeTop];
+    [_tableViewController.view autoPinToBottomLayoutGuideOfViewController:self withInset:0];
+    _tableViewController.tableView.tableHeaderView = searchBar;
+
+    _noSignalContactsView = [self createNoSignalContactsView];
+    self.noSignalContactsView.hidden = YES;
+    [self.view addSubview:self.noSignalContactsView];
+    [self.noSignalContactsView autoPinWidthToSuperview];
+    [self.noSignalContactsView autoPinEdgeToSuperviewEdge:ALEdgeTop];
+    [self.noSignalContactsView autoPinToBottomLayoutGuideOfViewController:self withInset:0];
+
+    [self updateTableContents];
+}
+
+- (UIView *)createNoSignalContactsView
```

Comment:
NBD. Just reducing code complexity a bit, and that it's somewhat redundant with the offers to invite contacts already in the view.